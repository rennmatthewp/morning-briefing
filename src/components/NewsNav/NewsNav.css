.nav {
  position: sticky;
  top: 0;
  height: 25px;
  width: 100%;
  padding: 5px 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #fff;
  overflow: hidden;
}

.nav-link {
  background-color: #f2f2f2;
  border-radius: 5px;
  width: fit-content;
  height: 20px;
  padding: 3px 5px 0;
}

.news-nav {
  border-bottom: 1px solid #000;
  margin-bottom: 5px;
}
