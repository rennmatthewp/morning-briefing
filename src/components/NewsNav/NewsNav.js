import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { newsSections } from '../../helper/apiCalls';
import './NewsNav.css';

export const NewsNav = ({ selectSection, category }) => {
  const sectionButtons = Object.keys(newsSections).map(section => {
    return (
      <NavLink
        to={`/${section}/${category}`}
        name={section}
        key={section}
        onClick={selectSection}
        className="nav-link">
        {newsSections[section]}
      </NavLink>
    );
  });

  return <div className="nav news-nav">{sectionButtons}</div>;
};

NewsNav.propTypes = {
  selectSection: PropTypes.func,
  category: PropTypes.string
};
