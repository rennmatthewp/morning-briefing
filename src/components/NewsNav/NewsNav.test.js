import React from 'react';
import { shallow } from 'enzyme';
import { NewsNav } from './NewsNav';

describe('NewsNav', () => {
  let renderedNewsNav;
  let mockSelectSection;

  beforeEach(() => {
    mockSelectSection = jest.fn();
    renderedNewsNav = shallow(<NewsNav selectSection={mockSelectSection} />);
  });

  it('should match the snapshot', () => {
    expect(renderedNewsNav).toMatchSnapshot();
  });

  it('should call selectSection onClick of a NavLink', () => {
    renderedNewsNav
      .find('NavLink')
      .first()
      .simulate('click');
    expect(mockSelectSection).toHaveBeenCalled();
  });
});
