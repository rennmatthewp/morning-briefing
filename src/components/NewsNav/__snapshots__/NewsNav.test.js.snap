// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsNav should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    selectSection={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-link"
          name="home"
          onClick={[Function]}
          to="/home/undefined"
>
          Home
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-link"
          name="world"
          onClick={[Function]}
          to="/world/undefined"
>
          World
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-link"
          name="national"
          onClick={[Function]}
          to="/national/undefined"
>
          National
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-link"
          name="nyregion"
          onClick={[Function]}
          to="/nyregion/undefined"
>
          NY Region
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-link"
          name="politics"
          onClick={[Function]}
          to="/politics/undefined"
>
          Politics
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-link"
          name="technology"
          onClick={[Function]}
          to="/technology/undefined"
>
          Technology
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-link"
          name="science"
          onClick={[Function]}
          to="/science/undefined"
>
          Science
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-link"
          name="business"
          onClick={[Function]}
          to="/business/undefined"
>
          Business
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-link"
          name="opinion"
          onClick={[Function]}
          to="/opinion/undefined"
>
          Opinion
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-link"
          name="tmagazine"
          onClick={[Function]}
          to="/tmagazine/undefined"
>
          NYT Magazine
</NavLink>,
      ],
      "className": "nav news-nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "home",
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Home",
          "className": "nav-link",
          "name": "home",
          "onClick": [Function],
          "to": "/home/undefined",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "world",
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "World",
          "className": "nav-link",
          "name": "world",
          "onClick": [Function],
          "to": "/world/undefined",
        },
        "ref": null,
        "rendered": "World",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "national",
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "National",
          "className": "nav-link",
          "name": "national",
          "onClick": [Function],
          "to": "/national/undefined",
        },
        "ref": null,
        "rendered": "National",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "nyregion",
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "NY Region",
          "className": "nav-link",
          "name": "nyregion",
          "onClick": [Function],
          "to": "/nyregion/undefined",
        },
        "ref": null,
        "rendered": "NY Region",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "politics",
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Politics",
          "className": "nav-link",
          "name": "politics",
          "onClick": [Function],
          "to": "/politics/undefined",
        },
        "ref": null,
        "rendered": "Politics",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "technology",
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Technology",
          "className": "nav-link",
          "name": "technology",
          "onClick": [Function],
          "to": "/technology/undefined",
        },
        "ref": null,
        "rendered": "Technology",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "science",
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Science",
          "className": "nav-link",
          "name": "science",
          "onClick": [Function],
          "to": "/science/undefined",
        },
        "ref": null,
        "rendered": "Science",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "business",
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Business",
          "className": "nav-link",
          "name": "business",
          "onClick": [Function],
          "to": "/business/undefined",
        },
        "ref": null,
        "rendered": "Business",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "opinion",
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Opinion",
          "className": "nav-link",
          "name": "opinion",
          "onClick": [Function],
          "to": "/opinion/undefined",
        },
        "ref": null,
        "rendered": "Opinion",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "tmagazine",
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "NYT Magazine",
          "className": "nav-link",
          "name": "tmagazine",
          "onClick": [Function],
          "to": "/tmagazine/undefined",
        },
        "ref": null,
        "rendered": "NYT Magazine",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            name="home"
            onClick={[Function]}
            to="/home/undefined"
>
            Home
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            name="world"
            onClick={[Function]}
            to="/world/undefined"
>
            World
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            name="national"
            onClick={[Function]}
            to="/national/undefined"
>
            National
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            name="nyregion"
            onClick={[Function]}
            to="/nyregion/undefined"
>
            NY Region
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            name="politics"
            onClick={[Function]}
            to="/politics/undefined"
>
            Politics
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            name="technology"
            onClick={[Function]}
            to="/technology/undefined"
>
            Technology
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            name="science"
            onClick={[Function]}
            to="/science/undefined"
>
            Science
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            name="business"
            onClick={[Function]}
            to="/business/undefined"
>
            Business
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            name="opinion"
            onClick={[Function]}
            to="/opinion/undefined"
>
            Opinion
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            name="tmagazine"
            onClick={[Function]}
            to="/tmagazine/undefined"
>
            NYT Magazine
</NavLink>,
        ],
        "className": "nav news-nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "home",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Home",
            "className": "nav-link",
            "name": "home",
            "onClick": [Function],
            "to": "/home/undefined",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "world",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "World",
            "className": "nav-link",
            "name": "world",
            "onClick": [Function],
            "to": "/world/undefined",
          },
          "ref": null,
          "rendered": "World",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "national",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "National",
            "className": "nav-link",
            "name": "national",
            "onClick": [Function],
            "to": "/national/undefined",
          },
          "ref": null,
          "rendered": "National",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "nyregion",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "NY Region",
            "className": "nav-link",
            "name": "nyregion",
            "onClick": [Function],
            "to": "/nyregion/undefined",
          },
          "ref": null,
          "rendered": "NY Region",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "politics",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Politics",
            "className": "nav-link",
            "name": "politics",
            "onClick": [Function],
            "to": "/politics/undefined",
          },
          "ref": null,
          "rendered": "Politics",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "technology",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Technology",
            "className": "nav-link",
            "name": "technology",
            "onClick": [Function],
            "to": "/technology/undefined",
          },
          "ref": null,
          "rendered": "Technology",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "science",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Science",
            "className": "nav-link",
            "name": "science",
            "onClick": [Function],
            "to": "/science/undefined",
          },
          "ref": null,
          "rendered": "Science",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "business",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Business",
            "className": "nav-link",
            "name": "business",
            "onClick": [Function],
            "to": "/business/undefined",
          },
          "ref": null,
          "rendered": "Business",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "opinion",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Opinion",
            "className": "nav-link",
            "name": "opinion",
            "onClick": [Function],
            "to": "/opinion/undefined",
          },
          "ref": null,
          "rendered": "Opinion",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "tmagazine",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "NYT Magazine",
            "className": "nav-link",
            "name": "tmagazine",
            "onClick": [Function],
            "to": "/tmagazine/undefined",
          },
          "ref": null,
          "rendered": "NYT Magazine",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
